ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -87,6 +89,38 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
             this.isScrolled = false;
         }
     }
+    // Navigate through fields
+    protected nextField(event: Event, {elements, preventDefault = false, carousel = false}: NavigateFields) {
+        preventDefault && event.preventDefault();
+        if(elements && elements.length > 0) {
+            // All provided elements array, excluding those, which doesn't have id
+            const elementsArray: Element[] = Array.from(document.querySelectorAll(elements.join(','))).filter(input => input.id && (input as HTMLElement)?.style.display !== 'none');
+            // Map data structure for elements in order to easily find them
+            const elementsObjects: Map<Element, number> = new Map(elementsArray.map((input, i) => [input, i]));
+            // Pressed element in a DOM
+            const pressedElement = event.target as Element;
+            // One-sided or Carousel behaviour while moving through fields
+            this.switchBehaviour(elementsArray, elementsObjects.get(pressedElement), carousel)
+        }
+    }
+    // Decide behaviour while navigation
+    private switchBehaviour(elements: Element[] | HTMLElement, elementIndex: number | undefined, carousel:boolean) {
+        if(carousel){
+            if(((elements as Element[]).length - 1 === elementIndex)) this.carouselRevert = true;
```

Comment:
I suppose in the described case above the focus will be moved back finally (instead of forward). 
Because there is a check for the first or the last index only. 
Is it ok?

Reply:
Yes, only in this scenario, where the pressed element is not in the elements array (like "body"). Also, I have discussed this particular case with QA, and he says it's fine. But there is always a different way. What would you suggest?